File: flutter_app/lib/features/orders/domain/repositories/order_repository.dart
import '../../../../core/result/result.dart';
import '../entities/order.dart';

abstract class OrderRepository {
  Future<Result<OrderEntity>> createOrder({
    required String customerId,
    required String serviceId,
    required String addressText,
  });

  Future<Result<OrderEntity>> acceptOrder(String orderId);
  Future<Result<OrderEntity>> startWork(String orderId);
  Future<Result<OrderEntity>> completeOrder(String orderId);
  Future<Result<OrderEntity>> rateOrder(String orderId, int rating);
}

File: apps/inwork_mobile/lib/features/orders/domain/order_repository.ts
export interface CreateOrderRequest {
  customerId: string;
  serviceId: string;
  addressText: string;
}

export interface OrderEntity {
  id: string;
  customerId: string;
  workerId?: string;
  serviceId: string;
  addressText: string;
  status: string;
  createdAt: string;
  updatedAt: string;
}

export interface OrderRepository {
  createOrder(request: CreateOrderRequest): Promise<OrderEntity>;
  acceptOrder(orderId: string): Promise<OrderEntity>;
  getOrder(orderId: string): Promise<OrderEntity>;
}
